dataStructures/queues/queue.go

```diff
@@ -39,52 +39,52 @@ func copyQ(q Queue) Queue {
 	return append(Queue{}, q...)
 }
 
-// Add method takes an order and adds it to the queue based on priority.
-func (q *Queue) Add(order *Order) {
-	if len(*q) == 0 {
-		*q = Queue{order}
-	} else {
-		var appended bool
-
-	QLoop:
-		for i, existingOrder := range *q {
-			var insert int
-
-			switch {
-			case order.priority < existingOrder.priority:
-				insert = i
-			case order.priority == existingOrder.priority:
-				insert = 1 + i
-				// peek to check next order in the queue.
-				if insert < len(*q) && (*q)[insert].priority == order.priority {
-					// keep looking for the best location to insert.
-					continue QLoop
-				}
-			default:
-				continue QLoop
+// binarySearch method will help to pick a slot to fix the next order
+// with complexity of Olog(n)
+func getInsertionSlot(q Queue, value int) int {
```

Comment:
Nit: prefixes like "get" or "handle" don't add anything to your function name so we recommend removing them. If it's a bit more descriptive like "fetch" (over the network / from a database) then it's OK.

Reply:
Nice!